File: boincsite/status/commandline/Task.py
import boincsite.status.Task as t


class Task(t.Task):

    def __init__(self, task):
        super().__init__(task)

        tasks = task.split('\n')

        line_mappings = {'name': 'name',
                         'WU name': 'workunit_name',
                         'project URL': 'project_url',
                         'report deadline': 'report_deadline',
                         'ready to report': 'ready_to_report',
                         'got server ack': 'got_server_ack',
                         'final CPU time': 'final_cpu_time',
                         'state': 'state',
                         'scheduler state': 'scheduler_state',
                         'exit_status': 'exit_status',
                         'signal': 'signal',
                         'suspended via GUI': 'suspended_via_gui',
                         'active_task_state': 'active_task_state',
                         'app version num': 'app_version_num',
                         'checkpoint CPU time': 'checkpoint_cpu_time',
                         'current CPU time': 'current_cpu_time',
                         'fraction done': 'fraction_done',
                         'swap size': 'swap_size',
                         'working set size': 'working_set_size',
                         'estimated CPU time remaining': 'estimated_cpu_time_remaining'}

        for ts in tasks:
            line_split = ts.split(':')
            line_split.reverse()
            key = line_split.pop().strip()
            if key in line_mappings:
                line_split.reverse()
                setattr(self, line_mappings[key], str.join(':', line_split).strip())

File: boincsite/zoetrope-client/src/components/task.js
import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { getTask, getProjects } from '../api/Api';
import { LabelValueRow } from './primitives/row';

const Task = () => {

  const { taskId } = useParams();

  const [task, setTask] = useState({});
  const [projects, setProjects] = useState([]);
  
  useEffect(() => {
    getTask(taskId, x => setTask(x));
    getProjects(x => setProjects(x));
  }, [taskId]);

  const getProject = () => 
    projects.filter(x => x.master_url===task.project_url)[0]?.project_name;

  return (
    <div>
      <h2>Task Summary: {taskId}</h2>
      <LabelValueRow label="Workunit name" value={task.name} />
      <LabelValueRow label="Project name" value={getProject()} />
      <LabelValueRow label="Percent done" value={task.fraction_done} />
      <LabelValueRow label="Time elapsed" value={task.final_cpu_time} />
      <LabelValueRow label="Time remaining" value={task.estimated_cpu_time_remaining} />
      <LabelValueRow label="Report deadline" value={task.report_deadline} />
      <LabelValueRow label="Ready to report?" value={task.ready_to_report?.toString()} />
      <LabelValueRow label="State" value={task.state} />
      <LabelValueRow label="Suspended via GUI?" value={task.suspended_via_gui?.toString()} />
      <LabelValueRow label="Active task state" value={task.active_task_state} />
      
    </div>
  );
};

export default Task;

